compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -1788,6 +1788,49 @@ impl<'a> Parser<'a> {
             self.sess.expr_parentheses_needed(&mut err, *sp);
         }
         err.span_label(span, "expected expression");
+        if self.prev_token.kind == TokenKind::BinOp(token::Plus)
```

Comment:
Can you maybe add a short comment here explaining what is done is this block, so that one can more easily skip reading this block?

Reply:
Or even better move it to its own method.

Reply:
It's in its own method (actually, multiple methods) now.